### PR TITLE
Ensure ServerChannel implementations accept multiple connections per …

### PR DESCRIPTION
**Pull request by normanmaurer**
…read loop.

Motivation:

Due how the DefaultMaxMessagesRecvByteBufAllocator is implemented we did only ever accept one connection per read loop even thought there might be more pending.

Modifications:

- Introduce ServerChannelRecvByteBufAllocator and use it for all our ServerChannel implementations

Result:

Fixes https://github.com/netty/netty/issues/11708

### CODE REVIEW

File: transport/src/main/java/io/netty/channel/ServerChannelRecvByteBufAllocator.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    public ServerChannelRecvByteBufAllocator() {
+        super(1, true);
+    }
+
+    @Override
+    public Handle newHandle() {
+        return new MaxMessageHandle() {
+            @Override
+            public int guess() {
+                return 128;
```
Comment (normanmaurer) [inline transport/src/main/java/io/netty/channel/ServerChannelRecvByteBufAllocator.java:31]: I just did pick something ... at the end it not matters much as we don't allocate `ByteBuf`.

File: pom.xml [hunk 2]
```diff
@@ -1033,6 +1033,7 @@
               <!-- Added to fix false-positive -->
               <exclude>io.netty.handler.codec.dns.TcpDnsQueryDecoder</exclude>
               <exclude>io.netty.handler.codec.dns.TcpDnsResponseEncoder</exclude>
+              <exclude>io.netty.channel.ServerChannelRecvByteBufAllocator</exclude>
```
Comment (normanmaurer) [inline pom.xml:1036]: This was needed :/

```
Error:  Failed to execute goal com.github.siom79.japicmp:japicmp-maven-plugin:0.15.3:cmp (default) on project netty-transport: There is at least one incompatibility: io.netty.channel.ServerChannelRecvByteBufAllocator:METHOD_ABSTRACT_ADDED_IN_IMPLEMENTED_INTERFACE -> [Help 1]
```

**Comment by chrisvest** [comment]
We only want to continue reading, when a message has been added, for acceptors?
I was wondering if any increment of the messages could be grounds for continuing.

Also, instead of adding all the constructor overloads, perhaps the `DefaultChannelConfig` could decide on what allocator to use by default, depending on whether the channel is a `ServerChannel` or not?

**Comment by normanmaurer** [comment]
> We only want to continue reading, when a message has been added, for acceptors? I was wondering if any increment of the messages could be grounds for continuing.

Yes only if the number of messages are smaller then the max number of messages. 
> 
> Also, instead of adding all the constructor overloads, perhaps the `DefaultChannelConfig` could decide on what allocator to use by default, depending on whether the channel is a `ServerChannel` or not?

I was thinking about this but was thinking that the way I did it here was less likely to break things for custom `Channel` implementations

File: transport/src/test/java/io/netty/channel/DefaultMaxMessagesRecvByteBufAllocatorTest.java [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+        handle.reset(channel.config());
+        handle.incMessagesRead(1);
+        handle.attemptedBytesRead(1);
+        handle.lastBytesRead(1);
+        assertTrue(handle.continueReading());
+        channel.finish();
+    }
+
+    @Test
+    public void testIngoreReadBytes() {
```
Comment (NiteshKant) [inline transport/src/test/java/io/netty/channel/DefaultMaxMessagesRecvByteBufAllocatorTest.java:59]: I think to justify the name of this test, you should add a call to `lastBytesRead(int bytes)` somewhere 😄 .. which also makes sense to check.

**Review by NiteshKant — APPROVED** [review]

**Review by chrisvest — APPROVED** [review]

**Pull request merged**